File: docs/examples/methodFunctions/toolsMethod/parse_array_code.rs
use lightvm::LightVM;
use lightvm::types::{capability::Capability, vmconfig::VmConfig};

fn main() {
  let mut vm = LightVM::new(VmConfig {
    caps: vec![Capability::Debug],
    ..Default::default()
  });
  let tools = vm.tools();

  let str_val = r#"
    push 5; ;; IP=0
    val x; ;; IP=1
    set x; ;; IP=2
  "#;

  let parsed = tools.parse_ltc_array(str_val);

  println!("{:#}", parsed.clone());
}

File: docs/examples/methodFunctions/toolsMethod/parseArrayCode.ts
import { Capability, LightVM } from 'lightvm';

const vm = new LightVM({ caps: [Capability.Debug] });
const tools = vm.tools();

const strVal = `
  push 5; ;; IP=0
  val x; ;; IP=1
  set x; ;; IP=2
`;
const parsed = tools.parseLTCArray(strVal);
console.log(parsed);
